File: my-folder/0392-is-subsequence/solution.cpp
class Solution
{
public:
    bool isSubsequence(string s, string t)
    {
        if (s.length() == 0)
        {
            return true;
        }
        else if (t.length() == 0)
        {
            return false;
        }
        int l = 0, r = 0;
        while (r < t.length() )
        {
            if (s[l] == t[r])
            {
                if (l == s.length() - 1)
                {
                    return true;
                }
                l++;
            }
            r++;
        }
        return false;
    }
};

File: my-folder/0392-is-subsequence/solution.php
class Solution {

    /**
     * @param String $s
     * @param String $t
     * @return Boolean
     */
    function isSubsequence($s, $t) {
        if(strlen($s)==0){
            return true;
        }
        else if(strlen($t)==0){
            return false;
        }
        $i = 0 ; $j = 0;
        while($j<strlen($t)){
            if($s[$i]==$t[$j]){
                if($i==strlen($s)-1){
                    return true;
                }
                $i++;
            }
            $j++;
        }
        return false;

    }
}
